src/vs/workbench/api/node/extHostSearch.ts

```diff
@@ -28,19 +29,22 @@ export class NativeExtHostSearch extends ExtHostSearch implements IDisposable {
 	private _internalFileSearchProvider: SearchService | null = null;
 
 	private _registeredEHSearchProvider = false;
+	private _numThreadsPromise: Promise<number | undefined> = Promise.resolve(undefined);
 
 	private readonly _disposables = new DisposableStore();
 
 	constructor(
 		@IExtHostRpcService extHostRpc: IExtHostRpcService,
 		@IExtHostInitDataService initData: IExtHostInitDataService,
 		@IURITransformerService _uriTransformer: IURITransformerService,
+		@IExtHostConfiguration _configurationService: IExtHostConfiguration,
 		@ILogService _logService: ILogService,
 	) {
 		super(extHostRpc, _uriTransformer, _logService);
 
 		const outputChannel = new OutputChannel('RipgrepSearchUD', this._logService);
-		this._disposables.add(this.registerTextSearchProvider(Schemas.vscodeUserData, new RipgrepSearchProvider(outputChannel)));
+		this._numThreadsPromise = _configurationService.getConfigProvider().then(provider => provider.getConfiguration('search').get<number>('ripgrep.numThreads'));
```

Comment:
I just tested this out, and it seems that it won't update correctly if I change the setting. This is probably because the configuration is initialized once here and isn't updated when changed. I think you should be able to listen to `onDidChangeConfiguration` on `ExtHostConfigProvider` to listen for this.

Reply:
Good point! I changed it to read the configuration value when the search starts so that always the latest configuration value is used.